presto-spi/src/main/java/com/facebook/presto/spi/block/MapBlockEncoding.java

```diff
@@ -82,10 +89,15 @@ public Block readBlock(BlockEncodingSerde blockEncodingSerde, SliceInput sliceIn
         Block keyBlock = blockEncodingSerde.readBlock(sliceInput);
         Block valueBlock = blockEncodingSerde.readBlock(sliceInput);
 
-        int[] hashTable = new int[sliceInput.readInt()];
-        sliceInput.readBytes(wrappedIntArray(hashTable));
+        int hashTableLength = sliceInput.readInt();
+        int[] hashTable = null;
+        if (hashTableLength >= 0) {
+            hashTable = new int[hashTableLength];
+            sliceInput.readBytes(wrappedIntArray(hashTable));
+        }
 
-        if (keyBlock.getPositionCount() != valueBlock.getPositionCount() || keyBlock.getPositionCount() * HASH_MULTIPLIER != hashTable.length) {
+        if (keyBlock.getPositionCount() != valueBlock.getPositionCount()
```

Comment:
please add clarifying parentheses

Reply:
Actually, I're prefer to see two separate checks. One to ensure the keys and values are the same position count, and one the verifies the hash table size.